compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -937,7 +937,7 @@ private BLangStatementExpression createIfElseFromConfigurable(BLangSimpleVariabl
         BLangStatementExpression stmtExpr = createStatementExpression(ifElse, resultVarRef);
         stmtExpr.setBType(configurableVar.getBType());
 
-        return rewrite(stmtExpr, initFunctionEnv);
+        return stmtExpr;
```

Comment:
Can we add a comment here explaining why we are not rewriting here?

Usually when we desugar stuff we desugar completely.

Reply:
Desugaring here is not the correct approach. We don't desugar any expressions of global-level variables here. The expressions of global-level variables are desugared when visiting the init function.

Do we need to add a special comment here?

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/42853/commits/b1e9ce64742fe61c6530a488e6377ed611b64a6d